### PR TITLE
chore(RHTAPWATCH-150): add go-mod-tidy

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
add go-mod-tidy check to the repository

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,3 +15,8 @@ repos:
         stages: [commit-msg]
     repo: https://github.com/jorisroovers/gitlint
     rev: v0.18.0
+  - hooks:
+      - id: go-mod-tidy
+    args: [--verbose]
```
Comment (yftacherzog) [inline .pre-commit-config.yaml:20]: Is it providing verbose output? I think the indentation is wrong.
```suggestion
  - hooks:
      - id: go-mod-tidy
        args: [--verbose]
```

**Review by yftacherzog — CHANGES_REQUESTED** [review]
Were you able to verify that it actually doing something? One way to go about it would be to duplicate one of the lines inside the `require` block in the `go.mod` file at the root of the repo. Running go mod tidy should eliminate the duplicate line, so the test should fail.

**Comment by ShakedAviv50313** [comment]
@yftacherzog I tried that and it did fail, now when it's fixed it is not failing anymore

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,3 +15,8 @@ repos:
         stages: [commit-msg]
     repo: https://github.com/jorisroovers/gitlint
     rev: v0.18.0
+  - hooks:
+      - id: go-mod-tidy
+        args: [--verbose]
```
Comment (yftacherzog) [inline .pre-commit-config.yaml:20]: Does this has any effect?

Comment (ShakedAviv50313) [reply .pre-commit-config.yaml]: apparently not, I removed it

Comment (yftacherzog) [reply .pre-commit-config.yaml]: The important bit is whether it makes a difference when there's an error. I think you should try to fail the check again and see if it makes a difference in that case.

Comment (ShakedAviv50313) [reply .pre-commit-config.yaml]: I failed the check a few times and it was the same with and without it

**Review by yftacherzog — APPROVED** [review]
lgtm

**Review by Omeramsc — APPROVED** [review]
LGTM

**Review by ifireball — APPROVED** [review]

**Pull request merged**